pkg/runtime/embedded.go

```diff
@@ -69,17 +69,23 @@ func (re *Unknown) UnmarshalJSON(in []byte) error {
 		return errors.New("runtime.Unknown: UnmarshalJSON on nil pointer")
 	}
 	re.TypeMeta = TypeMeta{}
-	re.RawJSON = append(re.RawJSON[0:0], in...)
+	re.Raw = append(re.Raw[0:0], in...)
+	re.ContentEncoding = ""
+	re.ContentType = ContentTypeJSON
 	return nil
 }
 
 // Marshal may get called on pointers or values, so implement MarshalJSON on value.
 // http://stackoverflow.com/questions/21390979/custom-marshaljson-never-gets-called-in-go
 func (re Unknown) MarshalJSON() ([]byte, error) {
-	if re.RawJSON == nil {
+	// If ContentType is unset, we assume this is JSON.
```

Comment:
I'll deal with this in the proposal, but I think internally (In code) empty contentType must mean JSON.  In serialized protobuf objects (from the outside) it means "protobuf encoding".  The code that creates Unknowns will have to get stricter - it may be that when we add the protobuf serializer (and unknown handling) we have to explicitly remove this check, and find all the places in the code where it breaks.  I'll do that as part of the serializer PR.